test/exports.spec.js

```diff
@@ -1,15 +1,14 @@
-import * as chai from 'chai'
-import * as path from 'path';
-import * as glob from 'glob';
-import toCamelCase from 'camelcase';
-import interopRequire from 'interop-require';
-import * as decorators from '../';
+const chai = require('chai');
+const path = require('path');
+const glob = require('glob');
+const camelCase = require('camelCase');
+const interopRequire = require('interop-require');
+const decorators = require('core-decorators');
```

Comment:
@jayphelps I had shared your hope, just want to make sure you understand that as of TypeScript@2.4, the limited support for property decorators means we can't really test _all_ of core-decorators from a TypeScript file.    This is the quote from the [TypeScript handbook ](https://www.typescriptlang.org/docs/handbook/decorators.html) I think reflects the limitation:

> NOTE  A Property Descriptor is not provided as an argument to a property decorator due to how property decorators are initialized in TypeScript. This is because there is currently no mechanism to describe an instance property when defining members of a prototype, and no way to observe or modify the initializer for a property. As such, a property decorator can only be used to observe that a property of a specific name has been declared for a class.

I've tried to find issues tracking this restriction, Microsoft/TypeScript#11866 seems one, but it's been closed, making reference to possibly reopening once the [Public class fields TC39 proposal](https://tc39.github.io/proposal-class-public-fields/) is ratified.   I don't fully understand that proposal, but it seems different from the stage 3 decorators one.   Can you see any reason they couldn't/shouldn't address it the way Babel does, by passing a synthetic (transpiler generated) descriptor to the first decorator in the list?